api/helpers.go

```diff
@@ -42,6 +43,26 @@ func FailWith(status int, message string, c echo.Context) error {
 	return c.JSON(status, payload)
 }
 
+// FailWithError fails with the specified error
+func FailWithError(err error, c echo.Context) error {
+	t := reflect.TypeOf(err)
```

Comment:
I am a bit afraid of reflections, but I agree it is the best choice here. We'll just need to keep track of the route latencies and resources usage after we release these changes.

Reply:
Alright! Since this is only called when an error occurs, and error rates for khan (New Relic) are really low, I don't expect the impact to be relevant. But we need to remember keeping track of this after deployment, sure.